common/src/main/java/com/wynntils/models/content/CaveModel.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.content;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Model;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.models.content.event.ContentUpdatedEvent;
+import com.wynntils.models.content.type.ContentInfo;
+import com.wynntils.models.content.type.ContentSortOrder;
+import com.wynntils.models.content.type.ContentType;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+public class CaveModel extends Model {
+    private List<CaveInfo> caves = new ArrayList<>();
+
+    public CaveModel(ContentModel contentModel) {
+        super(List.of(contentModel));
+    }
+
+    public void reloadCaves() {
+        WynntilsMod.info("Requesting rescan of caves in Content Book");
+        Models.Content.scanContentBook(ContentType.CAVE, this::updateCavesFromQuery);
+    }
+
+    private void updateCavesFromQuery(List<ContentInfo> newContent, List<StyledText> progress) {
```

Comment:
ahould we have a way to show the progress? we do for quests and discoveries...

Reply:
If there is a progress item in the content book, I will add that tomorrow.

Reply:
There is. All filters has a progress item. And the "All content" filter has a general overview of your character's progress. (We should probably parse this and display it nicely in the character view when we open the Wynntils book, i just realized)

Reply:
You can probably add a cave-"button" on the top left page, similar to the quest "button" we discussed earlier. It's probably not the best interface (not very intuitive!) but at least it is consistent with what we have.